warehouse/query-core/src/main/java/datawave/query/util/GeoUtils.java

```diff
@@ -0,0 +1,487 @@
+                    areaSortedRanges.add(biggestRange);
+                    positionSortedRanges.add(biggestRange);
+                    break;
+                }
+                
+                // break up the range, and only add the sub-ranges that intersect with the geometry
+                List<long[]> decomposedPositionRanges = decomposePositionRange(biggestRange.range[0], biggestRange.range[1]);
+                for (long[] range : decomposedPositionRanges) {
+                    Geometry rangeGeometry = positionRangeToGeometry(range[0], range[1]);
+                    if (rangeGeometry.intersects(geometry)) {
+                        double difference = areaDifference(rangeGeometry, geometry);
+                        RangeData rangeData = new RangeData(range, difference);
+                        areaSortedRanges.add(rangeData);
+                        positionSortedRanges.add(rangeData);
+                    }
+                }
+            }
+        }
+        
+        // return the discrete ranges after merging contiguous segments
+        return mergeContiguousRanges(positionSortedRanges);
+    }
+    
+    private static double areaDifference(Geometry geom1, Geometry geom2) {
+        try {
+            return geom1.getArea() - (geom1.intersection(geom2).getArea());
+        } catch (Exception e) {
+            log.warn("Unable to compute area difference between polygons. ", e);
+            if (log.isTraceEnabled()) {
+                log.trace("Geometries: [" + geom1.toText() + "], [" + geom2.toText() + "]");
+            }
+            return 0;
+        }
+    }
+    
+    /**
+     * Decomposes a range into its constituent ranges at the next lowest order of magnitude.
+     * 
+     * @param beginPosition
+     *            the minimum numeric index position
+     * @param endPosition
+     *            the maximum numeric index position
+     * @return a decomposed list of numeric index ranges
+     */
+    private static List<long[]> decomposePositionRange(long beginPosition, long endPosition) {
+        List<long[]> decomposedRanges = new ArrayList<>();
+        
+        long increment = (endPosition - beginPosition) / 10L;
+        if (increment > 1) {
+            for (long position = beginPosition; position < endPosition; position += increment) {
+                decomposedRanges.add(new long[] {position, position + increment});
+            }
+        } else {
+            decomposedRanges.add(new long[] {beginPosition, endPosition});
+        }
+        
+        return decomposedRanges;
+    }
+    
+    /**
+     * Converts a lat/lon pair a geo index
+     * 
+     * @param latitude
+     *            the latitude
+     * @param longitude
+     *            the longitude
+     * @return a geo index
+     */
+    public static String latLonToIndex(double latitude, double longitude) {
+        double latShift = latitude + 90.0;
+        double lonShift = longitude + 180.0;
```

Comment:
Consider adding constants for 90 degrees and 180 degrees.

Reply:
Fair enough.  These are only used here though, fwiw.